library-valcon-internship/src/components/AuthorForm/AuthorForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { Dispatch, FormEvent, SetStateAction } from 'react'
+
+import AuthorFormType from '../../models/AuthorFormType'
+import AuthorFormValidation from '../../models/AuthorFormValidation'
+import './AuthorForm.css'
+
+interface AuthorFormProps {
+  authorForm: AuthorFormType
+  authorFormValidation: AuthorFormValidation
+  setAuthorForm: Dispatch<SetStateAction<AuthorFormType>>
+  setAuthorFormValidation: Dispatch<SetStateAction<AuthorFormValidation>>
+  title?: string
+  onSubmit?: () => void
+}
+
+const AuthorForm = ({ authorForm, authorFormValidation,
+  setAuthorForm, setAuthorFormValidation, title, onSubmit }: AuthorFormProps) => {
+  const handleOnChangeInput = (value: string, formProperty: string) => {
+    switch (formProperty) {
+      case 'FirstName':
+        setAuthorForm((authorForm) => {
+          return {
+            ...authorForm,
+            firstName: value
+          }
+        })
+        break
+      case 'LastName':
+        setAuthorForm((authorForm) => {
+          return {
+            ...authorForm,
+            lastName: value
+          }
+        })
+        break
+      default:
+        break
+    }
+  }
```

Comment:
Could this be generic by renaming formProperty to firstName and just returning:
{
  ...authorFom,
  [formProperty]: value
}

Reply:
or by just lowercasing first letter of the key